core/src/main.cpp

```diff
@@ -547,15 +901,25 @@ auto main(int _argc, char* _argv[]) -> int
 		log::trace("Initializing listening socket (host=[{}], port=[{}]).", address.to_string(), port);
 		std::make_shared<listener>(ioc, tcp::endpoint{address, port}, config)->run();
 
-		// Capture SIGINT and SIGTERM to perform a clean shutdown.
+		// SIGINT and SIGTERM instruct the server to shut down.
+		// Ignore SIGPIPE. The iRODS networking code assumes SIGPIPE is ignored or caught.
 		log::trace("Initializing signal handlers.");
-		net::signal_set signals{ioc, SIGINT, SIGTERM};
-		signals.async_wait([&ioc](const beast::error_code&, int _signal) {
-			// Stop the io_context. This will cause run() to return immediately, eventually destroying the
-			// io_context and all of the sockets in it.
-			log::warn("Received signal [{}]. Shutting down.", _signal);
-			ioc.stop();
-		});
+		net::signal_set signals{ioc, SIGINT, SIGTERM, SIGPIPE};
+
+		const std::function<void(const beast::error_code&, int)> process_signals =
+			[&ioc, &signals, &process_signals](const beast::error_code&, int _signal) {
+				if (SIGPIPE == _signal) {
+					signals.async_wait(process_signals);
+					return;
+				}
+
+				// Stop the io_context. This will cause run() to return immediately, eventually destroying
+			    // the io_context and all of the sockets in it.
```

Comment:
indentation

Reply:
This is a weird clang-format thing. In github and my editor, every aligns correctly. Or maybe you caught the code before I ran clang-format?

Reply:
no?  i see it misaligned right here in github above where these words are appearing.

i see three tabs and four spaces, rather than four tabs, like the line above it.

Reply:
The clang-format workflow is happy. It's also aligned for this review comment. Is it misaligned here on your screen?

Reply:
clang-format is king.

but, yes, this is what's on my screen as i type here...   weird.

<img width="405" alt="whitespace" src="https://github.com/irods/irods_client_http_api/assets/55238/99e1ae98-e62e-4c9a-9b71-a840f2c9908d">

Reply:
I definitely seen that in tig. Looking at that same review comment shows the comment lines are aligned.

Not sure why we're seeing different things. Can screen share if that sounds helpful.

Reply:
nope, i'm good if clang-format is happy.    no need to debug github.